packages/graphback-core/src/runtime/CRUDService.ts

```diff
@@ -83,15 +83,19 @@ export class CRUDService<T = any> implements GraphbackCRUDService<T>  {
 
   //tslint:disable-next-line: no-any
   public async findBy(filter: any, context: GraphbackContext, orderBy?: GraphbackOrderBy, page?: GraphbackPage): Promise<ResultList<T>> {
-    let count: number;
-    if (context.graphback.options.aggregations?.count) {
-      count = await this.db.count(filter);
-    }
-
-    let items: T[];
-
-    if (context.graphback.options.selectedFields.length) {
+    let count: number; let items: T[];
+    const options = context.graphback.options;
+
+    if (options) { // query options defined, let's query db based on available informations
+      if (options.aggregations?.count) {
+        count = await this.db.count(filter);
+      }
+      if (options.selectedFields?.length) {
```

Comment:
This additional check seems redundant since it is checked in the data provider aswell:

https://github.com/aerogear/graphback/blob/4bd810161a09714e4d6cae7081518960b505e843/packages/graphback-runtime-knex/src/KnexDBDataProvider.ts#L120-L124

Reply:
> What happens if `options` is defined but `options.selectedFields?.length` is not?
> 
> it looks like `findBy` won't be called.

Yeap. This should be gone.